### PR TITLE
Adm 623[frontend]:add notifications when enter the export page and expired in 5 minutes

### PR DESCRIPTION
**Pull request by Andrea2000728**
## Summary

1.add notifications when enter the export page and expired in 5 minutes.
2. fix the header bar

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
image
image

## Note

_Null_

### CODE REVIEW

**Review by BoBoDai — APPROVED** [review]

File: frontend/__tests__/src/components/Common/NotificationButton/NotificationButton.test.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    expect(queryByTestId('NotificationIcon')).not.toBeInTheDocument()
+  })
+
+  it('should show NotificationPopper when clicking the component given the "open" value is true', async () => {
+    act(() => {
+      result.current.notificationProps = openNotificationProps
+    })
+    const { getByTestId, getByText } = render(<NotificationButton {...result.current} />)
+
+    fireEvent.click(getByTestId('NotificationIcon'))
```
Comment (guzhongren) [inline frontend/__tests__/src/components/Common/NotificationButton/NotificationButton.test.tsx:35]: should use `userEvent`, not `fireEventa`

File: frontend/__tests__/src/components/Common/NotificationButton/NotificationButton.test.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    expect(getByText('NotificationPopper')).toBeInTheDocument()
+  })
+
+  it('should hide NotificationPopper when clicking the component  given the "open" value is false', async () => {
+    act(() => {
+      result.current.notificationProps = closeNotificationProps
+    })
+    const { getByTestId, queryByText } = render(<NotificationButton {...result.current} />)
+
+    fireEvent.click(getByTestId('NotificationIcon'))
```
Comment (guzhongren) [inline frontend/__tests__/src/components/Common/NotificationButton/NotificationButton.test.tsx:45]: 1. same issue.
2. Extract `NotificationIcon` as a fixture or a global variable in this file

File: frontend/__tests__/src/components/Metrics/ReportStep/ReportStep.test.tsx [hunk 3]
```diff
@@ -246,4 +251,44 @@ describe('Report Step', () => {
       expect(getByTestId('loading-page')).toBeInTheDocument()
     })
   })
+
+  it('should call resetProps and updateProps when remaining time is less than or equal to 5 minutes', async () => {
+    const initExportValidityTimeMin = 30
+    React.useState = jest.fn().mockReturnValue([
+      initExportValidityTimeMin,
+      () => {
+        //do noting
```
Comment (guzhongren) [inline frontend/__tests__/src/components/Metrics/ReportStep/ReportStep.test.tsx:260]: `do noting`???
are u serious? mabey a mock function is better than current one

File: frontend/__tests__/src/hooks/useNotificationLayoutEffect.test.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { act, renderHook } from '@testing-library/react'
+import { useNotificationLayoutEffect } from '@src/hooks/useNotificationLayoutEffect'
+
+describe('useNotificationLayoutEffect', () => {
+  const defaultProps = { title: '', open: false }
+  it('should init the state of notificationProps when render hook', async () => {
+    const { result } = renderHook(() => useNotificationLayoutEffect())
+
+    expect(result.current.notificationProps).toEqual(defaultProps)
```
Comment (guzhongren) [inline frontend/__tests__/src/hooks/useNotificationLayoutEffect.test.tsx:9]: `notificationProps` is not a good name, `notification`

File: frontend/__tests__/src/hooks/useNotificationLayoutEffect.test.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  })
+
+  it('should reset the notificationProps when call resetProps given mock props', async () => {
+    const mockProps = { title: 'Test', open: true }
+    const { result } = renderHook(() => useNotificationLayoutEffect())
+    act(() => {
+      result.current.notificationProps = mockProps
+      result.current.resetProps?.()
```
Comment (guzhongren) [inline frontend/__tests__/src/hooks/useNotificationLayoutEffect.test.tsx:17]: 方法的执行导致了变量的值改变，不是变量也改变

File: frontend/src/components/Common/NotificationButton/NotificationButton.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { NotificationIconWrapper, sx } from '@src/components/Common/NotificationButton/style'
+import { ClickAwayListener, Tooltip } from '@mui/material'
+import { useNotificationLayoutEffectInterface } from '@src/hooks/useNotificationLayoutEffect'
+
+export const NotificationButton = ({ notificationProps, updateProps }: useNotificationLayoutEffectInterface) => {
+  const handleTooltipClose = () => {
+    updateProps!({
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/NotificationButton.tsx:7]: Forbidden non-null assertion

File: frontend/src/components/Common/NotificationButton/NotificationButton.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      title: notificationProps!.title,
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/NotificationButton.tsx:8]: Forbidden non-null assertion

File: frontend/src/components/Common/NotificationButton/NotificationButton.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+      open: false,
+    })
+  }
+  const toggleTooltip = () => {
+    updateProps!({
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/NotificationButton.tsx:13]: Forbidden non-null assertion

File: frontend/src/components/Common/NotificationButton/NotificationButton.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      title: notificationProps!.title,
+      open: !notificationProps!.open,
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/NotificationButton.tsx:15]: Forbidden non-null assertion

File: frontend/src/components/Common/NotificationButton/NotificationButton.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+export const NotificationButton = ({ notificationProps, updateProps }: useNotificationLayoutEffectInterface) => {
+  const handleTooltipClose = () => {
+    updateProps!({
+      title: notificationProps!.title,
+      open: false,
+    })
+  }
+  const toggleTooltip = () => {
+    updateProps!({
+      title: notificationProps!.title,
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/NotificationButton.tsx:14]: Forbidden non-null assertion

File: frontend/src/components/Common/NotificationButton/style.tsx [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { Z_INDEX } from '@src/constants'
+import { NotificationsRounded } from '@mui/icons-material'
+
+export const NotificationIconWrapper = styled(NotificationsRounded)({
+  color: theme.main.color,
+  marginRight: '0.5rem',
+})
+
+export const sx = {
+  width: '20vh',
```
Comment (guzhongren) [inline frontend/src/components/Common/NotificationButton/style.tsx:12]: `vh`?? our unit just only `rem`, follow our convention

File: frontend/src/components/HomeGuide/style.tsx [hunk 12]
```diff
@@ -1,5 +1,33 @@
...
+  '&:active': {
+    ...basicStyle,
+  },
+  '&:focus': {
+    ...basicStyle,
+  },
+})
+
+export const HomeGuideContainer = styled.div({
+  height: '80vh',
```
Comment (guzhongren) [inline frontend/src/components/HomeGuide/style.tsx:31]: change `vh` to `rem`

File: frontend/src/components/Metrics/ConfigStep/BranchSelection/index.tsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  const { id, organization, pipelineName, branches, onUpdatePipeline } = props
+  const branchesOptions: string[] = selectBranches(store.getState(), organization, pipelineName)
+  const isAllBranchesSelected = useMemo(
+    () => !_.isEmpty(branchesOptions) && _.isEqual(branches.length, branchesOptions.length),
+    [branches, branchesOptions]
+  )
+  const handleBranchChange = (event: React.SyntheticEvent, value: string[]) => {
+    let selectBranches = value
+    if (_.isEqual(selectBranches[selectBranches.length - 1], 'All')) {
+      /* istanbul ignore next */
```
Comment (guzhongren) [inline frontend/src/components/Metrics/ConfigStep/BranchSelection/index.tsx:26]: why ignore it!!!!!
PLZ fix it

File: frontend/src/components/Metrics/ReportStep/index.tsx [hunk 14]
```diff
@@ -184,11 +187,42 @@ const ReportStep = () => {
...
+
+  useLayoutEffect(() => {
+    if (exportValidityTimeMin) {
+      const startTime = Date.now()
+      const timer = setInterval(() => {
+        const currentTime = Date.now()
+        const elapsedTime = currentTime - startTime
+
+        const remainingTime = exportValidityTimeMin * 60 * 1000 - elapsedTime
+        if (remainingTime <= 5 * 60 * 1000) {
```
Comment (guzhongren) [inline frontend/src/components/Metrics/ReportStep/index.tsx:213]: give `5 * 60 * 1000` a readable variable name

File: frontend/src/theme.ts [hunk 15]
```diff
@@ -81,5 +84,8 @@ export const theme = createTheme({
     waringMessage: {
       color: '#cd5e32',
     },
+    tip: {
+      color: 'rgba(237,109,3,0.8)',
```
Comment (guzhongren) [inline frontend/src/theme.ts:88]: should not use `rgba`, use HEX

**Review by guzhongren — COMMENTED** [review]
Change with the comments

File: frontend/src/components/Common/NotificationButton/style.tsx (hunk 11 above)
Comment (guzhongren) [reply frontend/src/components/Common/NotificationButton/style.tsx:12]: focus this error

File: frontend/src/components/HomeGuide/style.tsx (hunk 12 above)
Comment (guzhongren) [reply frontend/src/components/HomeGuide/style.tsx:31]: same

**Review by guzhongren — COMMENTED** [review]
Change

**Pull request merged**